draft-ietf-tls-tls13.md

```diff
@@ -3980,16 +3983,20 @@ TLS-compliant application MUST implement the following TLS extensions:
   * Negotiated Groups ("supported_groups"; {{negotiated-groups}})
   * Key Share ("key_share"; {{key-share}})
   * Pre-Shared Key ("pre_shared_key"; {{pre-shared-key-extension}})
+  * Pre-Shared Key Exchange Modes ("psk_key_exchange_modes";
+    {{pre-shared-key-exchange-modes}})
   * Server Name Indication ("server_name"; Section 3 of {{RFC6066}})
   * Cookie ("cookie"; {{cookie}})
 
 All implementations MUST send and use these extensions when offering
 applicable cipher suites:
```

Comment:
Note that this isn't based on cipher suites any more, it's based on key exchange modes.  I would rearrange this section and say:

If you support Diffie-Hellman key exchange, you need key_share, supported_groups, and signature_algorithms.  If you support PSK, you need pre_shared_key, psk_key_exchange_modes and psk_binder.  It's also possible to support a a combination of both that excludes support for signature_algorithms if you only support DHE+PSK with PSK authentication...